src/app/dataset-block/metadata-block/components/event-details/components/set-polling-source-event/set-polling-source-event.component.html

```diff
@@ -0,0 +1,202 @@
+    </p>
+    <ng-container *ngIf="fetchStepUrl">
+        <p>
+            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;url:
+            <a [href]="fetchStepUrl.url" [title]="fetchStepUrl.url"
+                ><span class="value-color lh-sm">{{
+                    fetchStepUrl.url
+                }}</span></a
+            >
+        </p>
+        <p *ngIf="fetchStepUrl.eventTime">
+            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;eventTime:
+        </p>
+    </ng-container>
+    <ng-container *ngIf="isFetchStepFilesGlob">
+        <p>
+            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;path:
+            <span class="value-color">{{ fetchStepFilesGlob.path }}</span>
+        </p>
+    </ng-container>
+    <p class="color-key">read:</p>
+    <p>
+        &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;kind:
+        <span class="value-color">{{ event.read.__typename }}</span>
+    </p>
+    <ng-container *ngIf="isReadStepCsv">
+        <p>
+            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;separator:
+            <span class="value-color"> {{ readStepCsv.separator }}</span>
+        </p>
+        <p>
+            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;header:
+            <span class="value-color"> {{ readStepCsv.header }}</span>
+        </p>
+    </ng-container>
+    <p class="color-key">merge:</p>
+    <ng-container *ngIf="isMergeStrategyLedger">
+        <p>
+            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;kind:
+            <span class="value-color"> {{ event.merge.__typename }}</span>
+        </p>
+        <p>
+            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;primaryKey:
+            <span class="value-color">
+                {{ mergeStrategyLedger.primaryKey.join(", ") }}</span
+            >
+        </p>
+    </ng-container>
+
+    <ng-container *ngIf="event.preprocess">
+        <p class="color-key">preprocess:</p>
+        <p>
+            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;kind:
+            <span class="value-color"> {{ event.preprocess.__typename }}</span>
+        </p>
+
+        <p>
+            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;engine:
+            <span class="value-color"> {{ event.preprocess.engine }}</span>
+        </p>
+
+        <p *ngIf="event.preprocess.version">
+            &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;version:
+            <span class="value-color"> {{ event.preprocess.version }}</span>
+        </p>
+
+        <ng-container *ngIf="event.preprocess.queries">
+            <p>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;queries:</p>
+            <ng-container *ngFor="let query of event.preprocess.queries">
+                <p>
+                    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- alias:<span
+                        class="value-color"
+                    >
+                        {{ query.alias ? query.alias : "none" }}</span
+                    >
+                </p>
+                <p>
+                    &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; query:<span
+                        class="value-color lh-sm"
+                    >
+                        {{ query.query }}</span
```

Comment:
Query should be better formatted with SQL syntax highlighting.
Maybe we need a read-only Monaco component, or something simpler, as we don't need to edit.
This probably is out of scope for this ticket, still would be good to think how we are going to implement this.

Reply:
Added a read-only Monaco component